charts/templates/mongodb-svc.yaml

```diff
@@ -2,13 +2,15 @@
 {{ if or (and (hasKey .Values.replicaSet "extAccess") (hasKey .Values.replicaSet.extAccess "enabled") (eq .Values.replicaSet.extAccess.enabled true) (eq .Values.tls.enabled true)) (and (hasKey .Values.sharding "extAccess") (hasKey .Values.sharding.extAccess "enabled") (eq .Values.sharding.extAccess.enabled true)) -}}
 {{- $app := (.Values.clusterName | lower) -}}
 {{- $namespace := .Release.Namespace -}}
-{{- if and (hasKey .Values "sharding") (hasKey .Values.sharding "enabled") (eq .Values.sharding.enabled true) (hasKey .Values.sharding "extAccess") (hasKey .Values.sharding.extAccess "enabled") (eq .Values.sharding.extAccess.enabled true)}}
+{{- if and (hasKey .Values "sharding") (hasKey .Values.sharding "enabled") (eq .Values.sharding.enabled true) (hasKey .Values.sharding "extAccess") (hasKey .Values.sharding.extAccess "enabled") (eq .Values.sharding.extAccess.enabled true) (hasKey .Values.sharding.extAccess "ports")}}
 
+{{- $exposeMethod := ternary "NodePort" "LoadBalancer" (eq (.Values.replicaSet.extAccess.exposeMethod | lower) "nodeport")}}
+{{- range $index, $value := .Values.sharding.extAccess.ports }}
 ---
 apiVersion: v1
 kind: Service
 metadata:
-  name: {{ $app }}-svc-external
+  name: {{ $app }}-mongos-{{ $index }}-svc-external
```

Comment:
This template should be updated to: `{{ $app }}-sh-{{ $index }}-svc-external` to match what MEKO deploys.

Reply:
Thanks @slaskawi . We are attempting to only expose the mongos instances where, so should this therefore be `{{ $app }}-svc-external` as per https://www.mongodb.com/docs/kubernetes-operator/stable/tutorial/connect-from-outside-k8s/#create-an-external-service-for-the-mongos-pods ? (for sharded clusters)

Reply:
Yes, this should work fine.

You may also try out the `externalAccess` configuration: https://www.mongodb.com/docs/kubernetes-operator/master/tutorial/multi-cluster-connect-from-outside-k8s/index.html

It enables you to use either NodePort or LoadBalancer and provide the configuration via the CR.

Reply:
Thanks for confirming by Slack that `externalAccess` is more suited to LoadBalancer because we cannot specify all the ports required for the replica set.